Hero.cpp

```diff
@@ -45,10 +46,11 @@ void Hero::Attack(Monster &enemy)
             LevelUp();
         }
     }
-    if (damage - enemy.getDefense() > 0)
+    if (damage.physical - enemy.getDefense() > 0)
     {
-        xp += (damage - enemy.getDefense());
+        xp += (damage.physical - enemy.getDefense());
     }
+    xp += damage.magical;
```

Comment:
Az XP számlálás sem tökéletes, elméletielg csak azért a sebzésért jár XP, amit tényleg beütött.
Vagyis ha 20 ((physical - defense) + magic) sebzésünk van, van 10 HP-ja az ellenségnek, akkor az XP növekedés csak 10 lesz.